src/Pos/DB/Block.hs

```diff
@@ -225,7 +225,15 @@ delete :: (MonadDB m) => ByteString -> m ()
 delete k = rocksDelete k =<< getBlockIndexDB
 
 getData ::  (MonadIO m, MonadCatch m, Bi v) => FilePath -> m (Maybe v)
-getData fp = liftIO (rightToMaybe . decodeFull . BSL.fromStrict <$> BS.readFile fp)
+getData fp = flip catch handle $ liftIO $
+    either (\er -> throwM $ DBMalformed $ "Block deserialization failed: " <> show er) pure .
```

Comment:
Why block? AFAIU, we also store `Undo`s in files and this `getData` can be applied to such file.